src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -87,7 +87,9 @@ public class PowerShellFunctionInvoker : ScriptFunctionInvokerBase
                         System.Management.Automation.PowerShell.Create())
                 {
                     powerShellInstance.Runspace = runspace;
-                    _moduleFiles = GetModuleFilePaths(_host.ScriptConfig.RootScriptPath, _functionName);
+                    _moduleFiles = FindDuplicateModules(GetModuleFilePaths(_host.ScriptConfig.RootScriptPath, _functionName), _host.ScriptConfig.RootScriptPath, _functionName);
+                    // Remove duplicates from _moduleFiles and only keep the ones closest to the Function in moduleDirectory
+
                     if (_moduleFiles.Any())
                     {
                         powerShellInstance.AddCommand("Import-Module").AddArgument(_moduleFiles);
```

Comment:
@mathewc fair enough. If you like to cherry-pick the commits for the subdirectory code then I'm fine with that.
I'm also going to create a PR to remove auto-import altogether as I don't think that this is a great idea in general. This will address #897 then.